### PR TITLE
Support collection parameters in @Query methods

### LINKED ISSUE #1858: Collection parameters for @Query-annotated methods get escaped wrongly
The `@Query` annotation breaks when providing a collection parameter - it calls `toString()` on the parameter and escapes the result, which makes the list look like this `[\"some\",\"params\"]`, causing a 400 from Elasticsearch

### PR DESCRIPTION
**Pull request by herder**
This adds a test and fix so that collection parameters are supported and quoted correctly, at least for `Collection<String>`.

Previously the `StringQueryUtil` class would just call `.toString()` on the parameter and escape the `"` characters in the result, which would cause a syntax error in the query string.

This changes so that the class checks if the parameter is a `Collection` and if so starts a Json array and adds elements to it by calling `convert()` on each element.

Closes #1858 

- [x]  You have read the Spring Data contribution guidelines.
- [x]  There is a ticket in the bug tracker for the project in our issue tracker.
- [x]  You use the code formatters provided here and have them applied to your changes. Don’t submit any formatting related changes.
- [x]  You submit test cases (unit or integration tests) that back your changes.
- [x]  You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

**Review by sothawo — COMMENTED** [review]
Thanks a lot for this PR. This looks good, I just added a comment/question on this in the code

File: src/main/java/org/springframework/data/elasticsearch/repository/support/StringQueryUtil.java [hunk 1]
```diff
@@ -53,6 +56,22 @@ private static String getParameterWithIndex(ParameterAccessor accessor, int inde
...
+		return parameterValue;
+
+	}
+
+	private static String convert(Object parameter) {
+		if (Collection.class.isAssignableFrom(parameter.getClass())) {
+			Collection<?> collectionParam = (Collection<?>) parameter;
+			StringBuilder sb = new StringBuilder("[");
+			sb.append(collectionParam.stream().map(o -> "\"" + convert(o) + "\"").collect(Collectors.joining(",")));
+			sb.append("]");
```
Comment (sothawo) [inline src/main/java/org/springframework/data/elasticsearch/repository/support/StringQueryUtil.java:71]: This works, but what do you think about
```java
			StringBuilder sb = new StringBuilder("[");
			sb.append(collectionParam.stream().map(o -> {
				if (o instanceof String) {
					String s = (String) o;
					return '"' + convert(o) + '"';
				} else {
					return convert(o);
				}
			}).collect(Collectors.joining(",")));
			sb.append(']');
```
Might be better to only wrap Strings in quotes.

Comment (herder) [reply src/main/java/org/springframework/data/elasticsearch/repository/support/StringQueryUtil.java:71]: Yes, that makes sense. I'll add a test and incorporate your change.

Comment (herder) [reply src/main/java/org/springframework/data/elasticsearch/repository/support/StringQueryUtil.java]: Fixed in 8b154dfd , @sothawo

**Pull request merged**